src/server/scripts/Outland/zone_shadowmoon_valley.cpp

```diff
@@ -577,86 +589,105 @@ class npc_dragonmaw_peon : public CreatureScript
+            if (id == 1)
+            {
+                me->HandleEmoteCommand(EMOTE_ONESHOT_EAT);
+                events2.ScheduleEvent(EVENT_POISIONED, 5000);
+            }
+        }
 
-            if (id)
+        void CreditPlayer()
+        {
+            if (PlayerGUID)
             {
-                me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_EAT);
-                PoisonTimer = 15000;
-                SetCombatMovement(false);
+                Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID);
+                if (player && player->GetQuestStatus(11020) == QUEST_STATUS_INCOMPLETE)
+                    player->KilledMonsterCredit(23209, 0);
             }
         }
 
         void UpdateAI(uint32 diff)
         {
-            if (PoisonTimer)
+            events.Update(diff);
+            events2.Update(diff);
+
+            if (!UpdateVictim())
             {
-                if (PoisonTimer <= diff)
+                switch (events2.ExecuteEvent())
                 {
-                    if (PlayerGUID)
-                    {
-                        Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID);
-                        if (player && player->GetQuestStatus(11020) == QUEST_STATUS_INCOMPLETE)
-                            player->KilledMonsterCredit(23209, 0);
-                    }
-                    PoisonTimer = 0;
-                    Unit::DealDamage(me, me, me->GetHealth(), NULL, DIRECT_DAMAGE, SPELL_SCHOOL_MASK_NORMAL, NULL, false);
-                } else PoisonTimer -= diff;
 
+                case EVENT_WALK_TO_MUTTON:
+                    me->SetWalk(true);
+                    me->GetMotionMaster()->MovePoint(1, x, y, z, true);
+                    me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_STATE_NONE);
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_CHEER);
+                    break;
+                case EVENT_POISIONED:
+                    if (roll_chance_i(20))
+                        Talk(SAY_POISONED_1);
+                    me->CastSpell(me, SPELL_VOMIT);
+                    events2.ScheduleEvent(EVENT_KILL, 5000);
+                    break;
+                case EVENT_KILL:
+                    Unit::DealDamage(me, me, me->GetHealth(), NULL, DIRECT_DAMAGE, SPELL_SCHOOL_MASK_NORMAL, NULL, false);
+                    CreditPlayer();
+                    break;
+                }
                 return;
             }
 
-            if (!UpdateVictim())
-                return;
-            
-            if (KickTimer <= diff)
+            switch (events.ExecuteEvent())
             {
-                DoCastVictim(SPELL_KICK, false);
-                    KickTimer = 15000;
-            } else KickTimer -= diff;
 
-            if (SunderArmorTimer <= diff)
-            {
-                DoCastVictim(SPELL_SUNDER_ARMOR, false);
-                    SunderArmorTimer = 8000;
-            } else SunderArmorTimer -= diff;
+            case EVENT_KICK:
```

Comment:
Missing tab in here

Reply:
to me it looks like it's missing indentation too